main.go

```diff
@@ -38,22 +40,42 @@ func getMemcachedPods() (map[string]string, error) {
 		return nil, err
 	}
 
+	// a race exists here.  check to make sure that all the pods
+	// actually have an ip.  if they don't, pull the list again and check
```

Comment:
How about we filter on pods that return a status of Ready?  If a pod isn't passing its readiness probe, we want to ignore it, right?  I have a hunch that would also filter out the pods that are missing IP addresses.